extensions/amp-analytics/0.1/test/test-crc32.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {crc32} from '../crc32';
+
+const testVectors = [
+  {
+    input: '',
+    output: 0,
+  },
+  {
+    input: 'The quick brown fox jumps over the lazy dog',
+    output: 1095738169,
+  },
+  {
+    input: 'The quick brown fox jumps over the lazy dog.',
+    output: 1368401385,
+  },
+  {
+    input: 'hello',
+    output: 907060870,
+  },
+  {
+    input: 'world',
+    output: 980881731,
+  },
+  {
+    input: 'helloworld',
+    output: 4192936109,
+  },
+  {
+    input: '12345',
+    output: 3421846044,
+  },
+  {
+    input: '漢字',
+    output: 325357704,
```

Comment:
I don't think this is correct. CRC32 operates on bytes, not (UTF16) code points (Javascript strings).

You're going to have to convert from UTF16 code points to bytes, like https://github.com/SheetJS/js-crc32/blob/58bae10e50cdbbb945007baf1c0ca981756fbc13/crc32.js#L91-L107.

Reply:
Or, you can use our clever [`utf8Encode`](https://github.com/ampproject/amphtml/blob/464d59514d49f0b2cd08fdfc983beca83383a3c5/src/utils/bytes.js#L32-L42)